File: Sort.js
//Checks if the string contains number
function isNumeric(string){
    for(let i=0; i<string.length;i++){
        try{
            let num = parseInt(string[i]);
            if(!isNaN(num)) return true;
        }
        catch(error){
            console.log(string[i] + " Is not a number")
        }
    }
    return false;
}

//Split each letter and numbers(as one)
function SplitNum(string){
    string = string.toString().toLowerCase();
    let _string = [];
    let temp = "";
    let num = false;
    for(let i=0; i < string.length; i++){
        while(i < string.length){
            if(isNumeric(string[i])){
                temp += string[i];
                i++;
                num = true;
                continue;
            }
            else{
                if(num){
                    _string.push(temp);
                    temp = "";
                    num = false;
                }
                temp += string[i];
            }
            _string.push(temp);
            temp = "";
            break;
        }
        if(num){
            _string.push(temp);
            temp = "";
        }
    }
    return _string;
}


//Sorting based on the split letters and numbers
/**
 * @return {boolean}
 */
function Sort(a, b){
    a = SplitNum(a);
    b = SplitNum(b);
    let al = a.length;
    let bl = b.length;
    for(let i=0; i<(al < bl ? al : bl); i++){
        if(isNumeric(a[i]) && isNumeric(b[i])){
            if(parseInt(a[i]) !== parseInt(b[i])) return parseInt(a[i]) < parseInt(b[i]);
        }
        else if(!isNumeric(a[i]) && isNumeric(b[i])){
            return false;
        }
        else if(isNumeric(a[i]) && !isNumeric(b[i])){
            return true;
        }
        else if(!isNumeric(a[i]) && !isNumeric(b[i])){
            if(a[i] !== b[i]) return a[i] < b[i];

        }
    }
    return true;
}

//Sorting Call 
//Return sorted list 
//order(boolean):  true for "ASCENDING" and false for "DESCENDING"
function WindowsSort(list, order){
    let x = (order ? 1 : -1);
    let y = (!order ? 1 : -1);
    return list.sort((a,b) => {return Sort(a,b) ? x : y});
}

File: sort.py
def is_numeric(s):
    for char in s:
        try:
            num = int(char)
            if not num.isnumeric():
                return True
        except ValueError:
            print(char + " is not a number")
    return False

def split_num(s):
    s = str(s).lower()
    _string = []
    temp = ""
    num = False
    for char in s:
        if char.isnumeric():
            temp += char
            num = True
            continue
        else:
            if num:
                _string.append(temp)
                temp = ""
                num = False
            temp += char
        _string.append(temp)
        temp = ""
    if num:
        _string.append(temp)
    return _string

def sort_(a, b):
    a = split_num(a)
    b = split_num(b)
    al = len(a)
    bl = len(b)
    for i in range(min(al, bl)):
        if a[i].isnumeric() and b[i].isnumeric():
            if int(a[i]) != int(b[i]):
                return int(a[i]) < int(b[i])
        elif not a[i].isnumeric() and b[i].isnumeric():
            return False
        elif a[i].isnumeric() and not b[i].isnumeric():
            return True
        elif not a[i].isnumeric() and not b[i].isnumeric():
            if a[i] != b[i]:
                return a[i] < b[i]
    return True


def sort(a):
    a = split_num(a)
    length = len(a)
    lengthn = length + 1
    score = 0
    for i in range(length):
        x = i + 1
        val = int(a[i]) if a[i].isnumeric() else ord(a[i])
        score += (val * (lengthn - x))
    print("===========>Score: ", score, a)
    return score

def windows_sort(lst, reverse=False):
    return sorted(lst, key=lambda x: (sort(x), x), reverse=reverse)
